### PR TITLE
Admin stats

### PR DESCRIPTION
**Pull request by allenwq**
Provide more stats in system and instance admin pages:

- Users and Active users count
- Courses and Active courses count
- Active Course users tracking, timestamps are updated when users visit courses.

screenshots:
screen shot 2017-07-18 at 4 41 40 pm

screen shot 2017-07-18 at 4 48 07 pm

### CODE REVIEW

File: spec/models/course_spec.rb [hunk 1]
```diff
@@ -229,5 +229,38 @@
...
+        expect(new_count).to eq(old_count + 1)
+      end
+    end
+
+    describe 'calculated attributes' do
+      let(:course) { create(:course) }
+      let!(:course_users) { create_list(:course_user, 2, course: course) }
+
+      describe '.active_user_count' do
+        before { course_users.sample.update_column(:last_active_at, Time.zone.now) }
```
Comment (allenwq) [inline spec/models/course_spec.rb]: The rest of the volations won't fix, `update_column` gives much better performane than `save`

File: app/controllers/course/courses_controller.rb [hunk 2]
```diff
@@ -13,6 +13,8 @@ def show # :nodoc:
     @registration = Course::Registration.new
     @currently_active_announcements = current_course.announcements.currently_active
     @activity_feeds = recent_activity_feeds.limit(20).includes(activity: [:object, :actor])
+
+    current_course_user&.update_attribute(:last_active_at, Time.zone.now)
```
Comment (weiqingtoh) [inline app/controllers/course/courses_controller.rb]: Should we do this after all non-public actions that inherits `Course::Controller`?

File: app/views/system/admin/courses/_course.html.slim [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 = content_tag_for(:tr, course)
   td = (current_page - 1) * per_page + course_counter + 1
   th
     = link_to format_inline_text(course.title),
               course_url(course, host: course.instance.host, port: nil)
-  td = format_datetime(course.created_at)
+  td = format_datetime(course.created_at, :date_only_long)
+  td.active-level
+    div.progress
+      div.progress-bar.progress-bar-success style="width: #{active_level}%"
```
Comment (weiqingtoh) [inline app/views/system/admin/courses/_course.html.slim]: I think we can use the `display_progress_bar` helper here.

Comment (allenwq) [inline app/views/system/admin/courses/_course.html.slim]: Nice, updated !

File: app/controllers/course/courses_controller.rb (hunk 2 above)
Comment (allenwq) [inline app/controllers/course/courses_controller.rb]: In that case almost all of our actions (show/create/update/...) will involve this, which I think it's costly and unnecessary. 

I think current solution is good enough. But If we want more accurate results, the other way is to adopt this to the index action of each components.  Thoughts ?

**Comment by allenwq** [comment]
@weiqingtoh @kxmbrian I changed the implementation to update the timestamp every hour for all actions. It will give more accurate results and the performance overhead is minimized.

**Comment by fonglh** [comment]
Maybe add a tooltip on the active course users header to explain that it counts users who have accessed the course in the last 7 days?

**Comment by kxmbrian** [comment]
LGTM!

**Comment by allenwq** [comment]
@fonglh Tooltips added, thanks !

**Pull request merged**